### PR TITLE
add in game chat

### LINKED ISSUE #286: Implement in-game text chat
**Is your feature request related to a problem? Please describe.**
There is currently no way for players to communicate with each other while playing the game. It relies on third party group chats.

**Describe the solution you'd like**
A text chat while playing the game. This text chat allows everyone in the game to message everyone else 

**Additional context**
Here is a basic chat implementation to build off of: https://socket.io/get-started/chat

### Feature Approval:
- [x]  Feature approved by:
@hsingh124

### PR DESCRIPTION
**Pull request by hsingh124**
# Description
This PR fixes #286

Please include a summary of the changes here.
This PR adds in game chat for players. Before the chat functionality was only present for the new game screen.

# Demo

Before:
Screen Shot 2021-04-11 at 8 34 10 PM

After:
Screen Shot 2021-04-11 at 8 31 06 PM

# Type of change:
(Delete the ones that are not relevant)
* New feature (Non-breaking change that adds functionality)

# Checklist:
* Have you merged main into your branch?
* Have you tested your changes to ensure it works as expected and does not break existing functionality?
* If applicable, please ensure sufficient tests are added that is related to the changes.
* Please ensure this PR has a `label`, is linked to an `issue` and is related to a `project`
* If new documentation is required for this change, have you created a new documentation issue that describes the documentation needed?

### CODE REVIEW

**Review by flexzy — DISMISSED** [review]
I've tested it on my local machine and the in-game chat works great! Terrific job on this frontend addition. 

One thing that could be considered for the future is to redesign the in-game chat styling so that it fits a little better with the in-game theme, as well as those rounded corners. If you could create a new issue for it that would be great @hsingh124! 

For this current issue of just getting the functionality working, everything LGTM!

**Comment by JChamberlain28** [comment]
I cannot approve it as it is not my teams work. But its excellent to see this feature added. I think it adds real value as there was no way to communicate in game previously. This is essential for the working of Mafia. Reliance on Discord is not mandatory now. Good work!

File: frontend/src/Socket.js
Comment (JChamberlain28) [inline frontend/src/Socket.js]: I'm not really reviewing this but just spotted this. a2-main does not use the local server. Should this change be in a separate basic issue. It does not fit so much with this issue, and impacts all other PR's.

**Review by JChamberlain28 — COMMENTED** [review]
Please see my comment :)

**Review by cng822 — COMMENTED** [review]
The chat features looks absolutely amazing Harpreet! Thanks for doing this. It makes conversations in-game much easier now. 

I've noticed that depending on the browser size sometimes when you scroll there is a bit of whitespace at the bottom of the game chat, so it's not exactly aligned with the game itself. This is a simple fix/possible future issue and does not affect the functionality of the game.

File: frontend/src/Socket.js
Comment (hsingh124) [reply frontend/src/Socket.js]: Yeah, we had a discussion in the team 1 channel about this. We decided we should update it as we have to manually change to the dev server every time. I guess we could make a seperate issue for it.

**Comment by hsingh124** [comment]
> I've tested it on my local machine and the in-game chat works great! Terrific job on this frontend addition.
> 
> One thing that could be considered for the future is to redesign the in-game chat styling so that it fits a little better with the in-game theme, as well as those rounded corners. If you could create a new issue for it that would be great @hsingh124!
> 
> For this current issue of just getting the functionality working, everything LGTM!

Yup, will create a new issue for that, thanks! 🙂

**Comment by hsingh124** [comment]
> The chat features looks absolutely amazing Harpreet! Thanks for doing this. It makes conversations in-game much easier now.
> 
> I've noticed that depending on the browser size sometimes when you scroll there is a bit of whitespace at the bottom of the game chat, so it's not exactly aligned with the game itself. This is a simple fix/possible future issue and does not affect the functionality of the game.

Yeah, will create a new issue for it. Thanks! 🙂

**Review by cng822 — DISMISSED** [review]

**Review by flexzy — DISMISSED** [review]
Thanks for that quick change, no actual functionality has changed so LGTM!

**Review by cng822 — DISMISSED** [review]
LGTM!

**Review by a-158w — APPROVED** [review]
That rebase looks fine to me.

**Review by cng822 — APPROVED** [review]
Looks good. Thanks for rebasing :)

**Pull request merged**
